ec/ecresource/trafficfilterassocresource/delete.go

```diff
@@ -20,31 +20,35 @@ package trafficfilterassocresource
 import (
 	"context"
 	"errors"
-
-	"github.com/elastic/cloud-sdk-go/pkg/api"
 	"github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi/trafficfilterapi"
 	"github.com/elastic/cloud-sdk-go/pkg/client/deployments_traffic_filter"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
 )
 
-// delete will delete an existing deployment traffic filter ruleset association.
-func delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	var client = meta.(*api.API)
+func (t trafficFilterAssocResource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	var state modelV0
 
-	params := expand(d)
-	params.API = client
+	diags := request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
 
-	if err := trafficfilterapi.DeleteAssociation(trafficfilterapi.DeleteAssociationParams(params)); err != nil {
+	if err := trafficfilterapi.DeleteAssociation(trafficfilterapi.DeleteAssociationParams{
+		API:        t.provider.GetClient(),
+		ID:         state.TrafficFilterID.Value,
+		EntityID:   state.DeploymentID.Value,
+		EntityType: entityTypeDeployment,
+	}); err != nil {
 		if associationDeleted(err) {
-			d.SetId("")
-			return nil
+			response.State.RemoveResource(ctx)
+			return
 		}
-		return diag.FromErr(err)
+		response.Diagnostics.AddError(err.Error(), err.Error())
+		return
 	}
 
-	d.SetId("")
-	return nil
+	response.State.RemoveResource(ctx)
```

Comment:
There is a [statement](https://www.terraform.io/plugin/framework/migrating/resources/crud#migration-notes) that there is no need to call `response.State.RemoveResource()` explicitly in `Delete` though maybe it makes sense to call it  :).

Reply:
Thanks for the hint. I've updated/simplified the code accordingly. 👍

Reply:
Calling code:
```
	logging.FrameworkDebug(ctx, "Calling provider defined Resource Delete")
	resourceImpl.Delete(ctx, deleteReq, &deleteResp)
	logging.FrameworkDebug(ctx, "Called provider defined Resource Delete")

	if !deleteResp.Diagnostics.HasError() {
		logging.FrameworkTrace(ctx, "No provider defined Delete errors detected, ensuring State is cleared")
		deleteResp.State.RemoveResource(ctx)
	}
```